idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/folding/FoldingSyntax.kt

```diff
@@ -1,48 +1,35 @@
 package arrow.meta.ide.dsl.editor.folding
 
 import arrow.meta.ide.IdeMetaPlugin
+import arrow.meta.ide.phases.editor.extension.ExtensionProvider
 import arrow.meta.phases.ExtensionPhase
 import com.intellij.lang.ASTNode
 import com.intellij.lang.Language
 import com.intellij.lang.folding.FoldingBuilder
-import com.intellij.lang.folding.FoldingBuilderEx
 import com.intellij.lang.folding.FoldingDescriptor
-import com.intellij.lang.folding.LanguageFolding
 import com.intellij.openapi.editor.Document
-import com.intellij.psi.PsiElement
-import com.intellij.util.KeyedLazyInstance
 import org.jetbrains.kotlin.idea.KotlinLanguage
+import org.jetbrains.kotlin.psi.KtElement
 
 interface FoldingSyntax {
-  fun IdeMetaPlugin.addFoldingBuilder(
-    placeHolderText: (node: ASTNode) -> String?,
-    foldRegions: (root: PsiElement, document: Document, quick: Boolean) -> List<FoldingDescriptor>,
-    isCollapsedByDefault: (node: ASTNode) -> Boolean,
+
+  fun IdeMetaPlugin.registerFoldingBuilder(foldingBuilder: FoldingBuilder, lang: Language = KotlinLanguage.INSTANCE): ExtensionPhase =
+    ExtensionProvider.AddFoldingBuilder(lang, foldingBuilder)
+
+  fun FoldingSyntax.foldingBuilder(
+    placeHolderText: (ktElement: KtElement) -> String?,
```

Comment:
the first argument should not be changed

Reply:
why not? we are always using the ktElement, right?